### PR TITLE
[FIX] stock: speedup _get_inventory_lines_values

### PR DESCRIPTION
**Pull request by Aurelienvd**
When an inventory has lots of non-zero lines,
i.e. lines with non-exhausted products,
`action_start` is currently quite slow
because of the browse inside the for loop
of `_get_inventory_lines_values`.

Speeds this up by moving
the browse outside of the for loop.

(In recent PR odoo/odoo#79587 I only tested with exhausted inventories so I missed
a bottleneck when calling `action_start` on an inventory without exhausted products 😓)

#### speedup

| Number of lines | Before PR | After PR |
|:----------------:|:-----------:|:--------:|
|  10000 | 1m05s | 9.5s | 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_inventory.py [hunk 1]
```diff
@@ -313,8 +314,11 @@ def _get_inventory_lines_values(self):
                 'location_id': location_id,
                 'package_id': package_id
             }
-            line_values['product_uom_id'] = self.env['product.product'].browse(product_id).uom_id.id
+            product_ids.add(product_id)
             vals.append(line_values)
+        product_id_to_product = dict(zip(product_ids, self.env['product.product'].browse(product_ids)))
```
Comment (Whenrow) [inline addons/stock/models/stock_inventory.py:319]: I did not realize it in the 14.0 PR but are we sure the set order is preserved through the `zip()` ?

Comment (Aurelienvd) [reply addons/stock/models/stock_inventory.py:320]: Yeah you're right, I missed that. I replaced the set by an OrderedSet to ensure the order is preserved.

For the PR that got merged into 14.0 it should be fine as I'm zipping a recordset and a list
https://github.com/odoo/odoo/blob/26bdebf6fd8b77013ea449a02184e8a843ffe9e9/addons/stock/models/stock_inventory.py#L455-L456

But I also made that mistake in the 13.0 PR odoo/odoo#80106. Thankfully it has not been merged yet.

**Comment by Whenrow** [comment]
robodoo r+ 
fw-bot ignore

**Comment by Whenrow** [comment]
@Aurelienvd Thanks

**Pull request closed without merging**